### PR TITLE
Add "preview features" config switch

### PR DESCRIPTION
**Pull request by MattIrv**
Adds a config switch to disable preview features. Users are prompted on first launch to enable preview features, and otherwise can set it via the regular config.

Currently the features marked as preview are backup, restore, chart viewer, query plan, and extension-contributed dashboards. If the preview features config is disabled then all except backup and restore will not show up at all. Backup and restore will show a pop up when you try to use them telling you to enable the feature

Prompt for users to enable preview features

### CODE REVIEW

**Comment by kburtram** [comment]
@MattIrv can we iterate with PM on the specific message text.  I think something more simpler like "Would you like to enable preview features? Yes \ No" may be sufficient and not require updating the string as features move in and out of preview state.

**Comment by MattIrv** [comment]
@yualan Any thoughts on the proposed language Karl put above?

**Comment by yualan** [comment]
I will review this with @vickyharp this week. Ideally, it would make sense that:

- User knowingly chooses to see preview features or not
- User knows what features are considered preview.

We should make sure that if the user quickly X's out of the pop-up, that it doesn't auto-select opting in or out. The pop-up should persist on start-up until the user concretely clicks yes or no.

I agree with Karl's suggestion, but we should also be clear of which features are considered preview so that user can decide if they really want to disable viewing preview features. The main issue is that if users perceive that preview features aren't worth using in the short-term, the setting persists across releases so they might not see other preview features that they might actually want to see.

We can close on this this week, thanks

File: src/sql/parts/query/editor/queryEditor.ts [hunk 1]
```diff
@@ -461,6 +463,16 @@ export class QueryEditor extends BaseEditor {
 		this._estimatedQueryPlanAction = this._instantiationService.createInstance(EstimatedQueryPlanAction, this);
 		this._actualQueryPlanAction = this._instantiationService.createInstance(ActualQueryPlanAction, this);
 
+		this.setTaskbarContent();
+
+		this._configurationService.onDidChangeConfiguration(e => {
```
Comment (anthonydresser) [inline src/sql/parts/query/editor/queryEditor.ts]: Some places you listen on configuration changes and some places you don't, would it be better to just reload the window when the user selects to enable preview features?

**Comment by kburtram** [comment]
@MattIrv  can we try to merge this today if it's ready?  We need this for the next test pass.

BTW, it looks like VS Code is using the below menu for "Never Show Again".  I think the current approach with the 3rd button is fine, but it may be slightly better to use the standard mechanism for this.

image

**Comment by MattIrv** [comment]
It’s already using that Dropdown for the never show again option but I’ll update the exact wording of the choice to match what VS Code is doing too

**Comment by kburtram** [comment]
@MattIrv thanks!  The wording is fine with me either way 😄

File: src/sql/parts/query/editor/queryEditor.ts (hunk 1 above)
Comment (MattIrv) [reply src/sql/parts/query/editor/queryEditor.ts]: You’re right that some places don’t watch the config changes, for example users would have to close and reopen a dashboard after making the change in order to see extension tabs, but I don’t think it’s going to be a super common scenario so I won’t worry about it too much.

**Pull request merged**